custom_components/waste_collection_schedule/waste_collection_schedule/source/kingston_gov_uk.py

```diff
@@ -39,6 +39,16 @@ class Source:
     def __init__(self, uprn: str):
         self._uprn = str(uprn)
 
+    def interpret_date(self, date_string):
+        now = datetime.now()
+        date = datetime.strptime(date_string + f" {now.year}", "%A %d %B %Y")
+
+        if date < now:
```

Comment:
This is also true for collections on the current date so they will be moved to the next year. You could resolve this by converting `now` and `date` from `datetime` objects to `date` objects before this instead of at the end of the function.